### PR TITLE
C# support, RemoteTestHarness, UWP fixes

### PR DESCRIPTION
**Pull request by Sanakan8472**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,11 +14,20 @@ ez_detect_project_name(DETECTED_NAME)
...
-	DESCRIPTION "An open-source game engine"
-	HOMEPAGE_URL "http://ezEngine.net"
-	LANGUAGES C CXX)
+if (CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
+	# C# support only on windows desktop.
+	project (${EZ_SOLUTION_NAME}
+		VERSION 1.0
+		DESCRIPTION "An open-source game engine"
+		HOMEPAGE_URL "http://ezEngine.net"
+		LANGUAGES C CXX CSharp)
```
Comment (jankrassnigg) [inline CMakeLists.txt]: You could probably put the languages into a variable.

File: Code/BuildSystem/CMake/ezUtilsNuGet.cmake [hunk 2]
```diff
@@ -0,0 +1,24 @@
+######################################
+### NuGet support
+######################################
+
+######################################
+### ez_requires_nuget()
+######################################
+
+function(ez_requires_nuget)
```
Comment (jankrassnigg) [inline Code/BuildSystem/CMake/ezUtilsNuGet.cmake]: My "requires" functions meant to exclude the project, if X is not available.
Could you rename this?

File: Code/BuildSystem/CMake/ezUtilsTargetCS.cmake [hunk 3]
```diff
@@ -0,0 +1,61 @@
+######################################
+### ez_create_target_cs(<LIBRARY | APPLICATION> <target-name>)
+######################################
+
+function(ez_create_target_cs TYPE TARGET_NAME)
+
+    set(ARG_OPTIONS NO_PCH NO_UNITY NO_QT NO_EZ_PREFIX)
```
Comment (jankrassnigg) [inline Code/BuildSystem/CMake/ezUtilsTargetCS.cmake]: Looks like you are not using most of these parameter options

File: CMakeLists.txt (hunk 1 above)
Comment (Sanakan8472) [reply CMakeLists.txt]: Done!

File: Code/BuildSystem/CMake/ezUtilsTargetCS.cmake (hunk 3 above)
Comment (Sanakan8472) [reply Code/BuildSystem/CMake/ezUtilsTargetCS.cmake]: Yep, did some cleanup.

File: Code/BuildSystem/CMake/ezUtilsNuGet.cmake (hunk 2 above)
Comment (Sanakan8472) [reply Code/BuildSystem/CMake/ezUtilsNuGet.cmake]: Good point, renamed it to ez_nuget_init(). The function should never fail unless MS server goes down or you don't have internet.

**Pull request merged**